java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll

```diff
@@ -167,14 +167,15 @@ class FrameworkModeMetadataExtractor extends MetadataExtractor {
 
   override predicate hasMetadata(
     Endpoint e, string package, string type, boolean subtypes, string name, string signature,
-    int input
+    int input, string parameterName
   ) {
     exists(Callable callable |
       e.asParameter() = callable.getParameter(input) and
       package = callable.getDeclaringType().getPackage().getName() and
       type = callable.getDeclaringType().getErasure().(RefType).nestedName() and
       subtypes = this.considerSubtypes(callable) and
-      name = e.toString() and
+      name = callable.getName() and
+      parameterName = e.asParameter().getName() and
```

Comment:
Note that this forces `e` to always be a `Parameter`, and `hasMetadata` will not hold if it's anything else. I guess this is fine in framework mode, but I just wanted to mention it.

Reply:
This is a good point. Perhaps instead we should add a `getName` method to the interface for `EndPoint`, and call that. Internally (for the version for which  `EndPoint` is `ParameterNode`) we can then call `.asParameter().getName()`.

Reply:
> I guess this is fine in framework mode, but I just wanted to mention it.

I think that's completely fine here! In application mode, we wouldn't export the method parameter at all (it's not available there)